File: app/src/main/java/de/westnordost/streetcomplete/quests/PriorityList.java
package de.westnordost.streetcomplete.quests;

import java.util.List;

import de.westnordost.streetcomplete.view.Item;

public class PriorityList {
    public static List<Item> buildList(List<Item> allItems, List<String> popularNames){
        // in reverse because the first element in the list should be first in religionsList
        for (int i = popularNames.size()-1; i >= 0; --i)
        {
            String popularReligionName = popularNames.get(i);
            for(int j = 0; j < allItems.size(); ++j)
            {
                Item processed = allItems.get(j);
                if(processed.value.equals(popularReligionName))
                {
                    // shuffle to start of list
                    allItems.remove(j);
                    allItems.add(0,processed);
                    break;
                }
            }
        }
        return allItems;
    }
}

File: app/src/main/java/de/westnordost/streetcomplete/quests/PriorityList.kt
package de.westnordost.streetcomplete.quests

import de.westnordost.streetcomplete.view.Item

fun Collection<Item<String>>.sortedBy(names: Iterable<String>): List<Item<String>> {
    val result = toMutableList()
    // in reverse because the first element in the list should be first in religionsList
    for (name in names.reversed()) {
        for (i in result.indices) {
            val processed = result[i]
            if (processed.value == name) {
                // shuffle to start of list
                result.removeAt(i)
                result.add(0, processed)
                break
            }
        }
    }
    return result
}

